OsmAnd/src/net/osmand/plus/measurementtool/adapter/MeasurementToolAdapter.java

```diff
@@ -88,15 +90,27 @@ public boolean onTouch(View view, MotionEvent event) {
 		if (!TextUtils.isEmpty(pointDesc)) {
 			holder.descr.setText(pointDesc);
 		} else {
+			String azimuth;
+			String text = "";
 			if (pos < 1) {
-				holder.descr.setText(mapActivity.getString(R.string.shared_string_control_start));
+				text = mapActivity.getString(R.string.shared_string_control_start);
+				Location l1 = mapActivity.getMyApplication().getLocationProvider().getLastStaleKnownLocation();
+				if (mapActivity.getMyApplication().getLocationProvider().getLastKnownLocation() != null) {
+					azimuth = OsmAndFormatter.getFormattedAzimuth(MapUtils.getBearingToPoint(l1, points.get(0).lat, points.get(0).lon));
```

Comment:
No distance?

Reply:
Yes, it's the first point. Azimuth there is calculated from your current location